tests/unit/processor/pseudonymizer/test_pseudonymizer.py

```diff
@@ -4,26 +4,738 @@
+        {
+            "filter": "pseudo_this",
+            "pseudonymizer": {
+                "pseudonyms": {
+                    "pseudo_this": "RE_WHOLE_FIELD",
+                }
+            },
+        },
+        {
+            "pseudo_this": ["foo"],
+        },
+        {
+            "pseudo_this": [
+                "<pseudonym:e008abcd3e050a10853e0c5f694a10e87d693b8cfdb3457e42376cb06ab218ed>"
+            ],
+        },
+        None,
+    ),
+    (
+        "pseudonymize_list_with_two_equal_element",
+        {
+            "filter": "pseudo_this",
+            "pseudonymizer": {
+                "pseudonyms": {
+                    "pseudo_this": "RE_WHOLE_FIELD",
+                }
+            },
+        },
+        {
+            "pseudo_this": ["foo", "foo"],
+        },
+        {
+            "pseudo_this": [
+                "<pseudonym:e008abcd3e050a10853e0c5f694a10e87d693b8cfdb3457e42376cb06ab218ed>",
+                "<pseudonym:e008abcd3e050a10853e0c5f694a10e87d693b8cfdb3457e42376cb06ab218ed>",
+            ],
+        },
+        None,
+    ),
+    (
+        "pseudonymize_list_with_two_different_element",
+        {
+            "filter": "pseudo_this",
+            "pseudonymizer": {
+                "pseudonyms": {
+                    "pseudo_this": "RE_WHOLE_FIELD",
+                }
+            },
+        },
+        {
+            "pseudo_this": ["foo", "bar"],
+        },
+        {
+            "pseudo_this": [
+                "<pseudonym:e008abcd3e050a10853e0c5f694a10e87d693b8cfdb3457e42376cb06ab218ed>",
+                "<pseudonym:98b611cbecbd6a4533695fad8b40a46210f736ae3ef450fb9c4ab65638397113>",
+            ],
+        },
+        None,
+    ),
+    (
+        "pseudonymize_one_element_from_list_with_two_different_elements",
+        {
+            "filter": "pseudo_this",
+            "pseudonymizer": {
+                "pseudonyms": {
+                    "pseudo_this": "RE_DOMAIN_BACKSLASH_USERNAME",
+                }
+            },
+        },
+        {
+            "pseudo_this": ["foo\\test", "bar"],
+        },
+        {
+            "pseudo_this": [
+                "foo\\<pseudonym:d95ac3629be3245d3f5e836c059516ad04081d513d2888f546b783d178b02e5a>",
+                "bar",
+            ],
+        },
+        None,
+    ),
+]
+
+failure_test_cases = [  # testcase, rule, event, expected
+    (
+        "do_not_pseudonymize_url",
+        {
+            "filter": "filter_this: does_not_matter",
+            "pseudonymizer": {
+                "pseudonyms": {"pseudo_this": "RE_ALL_NO_CAP"},
+                "url_fields": ["pseudo_this"],
+            },
+        },
+        {
+            "filter_this": "does_not_matter",
+            "pseudo_this": "https://test.de",
+        },
+        {
+            "filter_this": "does_not_matter",
+            "pseudo_this": "https://test.de",
+        },
+        "tests/testdata/unit/pseudonymizer/pseudonymizer_regex_mapping.yml",
+    ),
```

Comment:
why is this a failure test case? And usually failure_test_cases have an exception to test for, don't they? Furthermore, I do not see any test where this list of failure test cases is actually being used.

Reply:
fragments from former iteration. deleted it.